### PR TITLE
Recording page

### PR DESCRIPTION
**Pull request by Alec-Hallman**
### Created the recording page. 

Users can click the microphone button to start a recording, clicking it again will pause the recording and display a little "saved" icon at the bottom of the screen. Whilst audio is being recorded, a sound wave will appear in the blue box displaying their mic input and the top box shows a sound speed clock as well as the recording status. When the recording page or app is closed, the audio file is accessible in your phone's download folder. The name of the file correlates to the date and time at which the recording was started. This page also features the bottom navigation bar (thanks Anika) and can traverse back and forth between the home page. A little "are you sure?" pop up will appear if the user tries to go to the homepage while recording.

> Notes
The recording page does not feature a stop/save button, as the audio recorder will automatically do so. Instead, I created a little pop-up to communicate when the recording is saved. Any feedback on this would be appreciated, as this still might not be as intuitive as it could be. But I enjoy how un intrusive this system is, require the user to press less buttons.

image

### CODE REVIEW

File: mobile/voice_journal_app/android/app/src/debug/AndroidManifest.xml
Comment (pabloj2001) [inline mobile/voice_journal_app/android/app/src/debug/AndroidManifest.xml]: While declaring permissions here works for Android, we won't get the permission on iOS devices. A better way to do it is to ask for permissions through Flutter when the record button is clicked: https://stackoverflow.com/a/63480023/4749973

File: mobile/voice_journal_app/lib/RecordingPage.dart
Comment (pabloj2001) [inline mobile/voice_journal_app/lib/RecordingPage.dart]: We generally want to avoid having duplicates of code in our project and instead want to reuse as much as possible. Instead of putting the bottom nav here, we'll want to put it in the main file and show the pages as widgets within it.

File: mobile/voice_journal_app/lib/home.dart
Comment (pabloj2001) [inline mobile/voice_journal_app/lib/home.dart]: We seem to be merging changes from other PRs here, this is generally bad practice. I know we need to update this page to add the navigation to the recording page, but let's hold off on it until Anika's home page changes get merged. Let's remove this page for now. Sorry if I made it confusing!

File: mobile/voice_journal_app/lib/Pages/Alec_Recording.dart
Comment (pabloj2001) [inline mobile/voice_journal_app/lib/Pages/Alec_Recording.dart]: I don't think we need another page as a starting point for the app. Let's remove this page if there's no need for it.

File: mobile/voice_journal_app/lib/RecordingPage.dart
Comment (pabloj2001) [inline mobile/voice_journal_app/lib/RecordingPage.dart]: There's no need for this starting point in the app since we already have `main.dart` and `RecordingPage` will be called when navigated to.

**Review by pabloj2001 — CHANGES_REQUESTED** [review]
Good job on this first PR! There's a few things I think we need to update but we're on the right track.

**Review by pabloj2001 — APPROVED** [review]
Looks Good to Me!

File: mobile/voice_journal_app/lib/RecordingPage.dart
Comment (anikapatychuk) [inline mobile/voice_journal_app/lib/RecordingPage.dart:1]: not sure if I looked at this at all right but I had a problem were as soon I hit record it paused me and the wouldn't un-pause

**Comment by TaseskiCS** [comment]
Looks like the functionality works smoothly, great work

**Review by TaseskiCS — COMMENTED** [review]
Just tested it, also had the unpause issue, but other than that everything works great!

**Pull request merged**
